[Taha Jailani] Duke Increments

PhireHandy opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Add configuration for console applications

Gradle defaults to an empty stdin which results in runtime exceptions
when attempting to read from `System.in`. Let's add some sensible
defaults for students who may still need to work with the standard
input stream.

j-lum committed:
Merge pull request #13 from j-lum/javaexec

Add configuration for console applications

PhireHandy committed:
Creation of MainManager class

PhireHandy committed:
Level-2

PhireHandy committed:
Task Class created, Level-3

PhireHandy committed:
Implemented Enums in Task Class

PhireHandy committed:
Level-4

PhireHandy committed:
Implementation of TaskException

PhireHandy committed:
DoneException + UnknownCommandException implemented, Level-5

PhireHandy committed:
Delete + DeleteException implemented, Level-6

PhireHandy committed:
Implemented saving to saving to/reading from disk + Level-7

PhireHandy committed:
Revert "Implemented saving to saving to/reading from disk + Level-7"

This reverts commit cb85e8d126c0c29e5dd9e6e17a0fc845714f87b5.

PhireHandy committed:
Implemented saving to file + Level-7

PhireHandy committed:
Implemented Date/Time + Level-8

PhireHandy committed:
Merge branch 'Branch-Level-7'

PhireHandy committed:
Merge branch 'Branch-Level-8'

PhireHandy committed:
Cleared the clashes from merging

damithc commented:
Please rectify the PR title to match the requirement. The PR author can edit the PR title.

PhireHandy committed:
A-MoreOOP

PhireHandy committed:
A-Packages

PhireHandy committed:
updated readme

PhireHandy committed:
committing A-Packages correctly

PhireHandy committed:
Merge branch 'gradle'
adding Gradle to project

PhireHandy committed:
implemented basic Gradle, Shadowjar and JUnit
requires further implementation with regards to testing (JUnit)

PhireHandy committed:
deleted an unnecessary System.out.println

PhireHandy committed:
implemented A-Jar through GitHub directly

PhireHandy committed:
Implemented A-JUnit testing for all Task classes and TaskList class

PhireHandy committed:
Implemened A-JavaDoc

PhireHandy committed:
Implemented A-CodingStyle and A-CheckStyle via Gradle

PhireHandy committed:
Implemented Level-9

PhireHandy committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/command/DeadlineCommand.java
#	src/main/java/command/DeleteCommand.java
#	src/main/java/command/DoneCommand.java
#	src/main/java/command/EventCommand.java
#	src/main/java/exception/DukeException.java
#	src/main/java/main/Storage.java
#	src/main/java/main/UI.java
#	src/main/java/task/Event.java
#	src/main/java/task/Todo.java

PhireHandy committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/main/UI.java
#	src/main/java/task/Task.java

PhireHandy committed:
Optimised FindCommand

PhireHandy committed:
Implemented JavaFX GUI

To be implemented:
Full support of all Duke's previous features with the new GUI

PhireHandy committed:
Implemented Duke backend communication with GUI frontend

PhireHandy committed:
Ensured the Test Classes follow the CheckStyle.

PhireHandy committed:
Implemented assertion checks in Parser.java and ensured that all files in Main directory follow the CheckStyle. (#5)

Assertion checks to ensure that the correct types of command are passed as inputs.

PhireHandy committed:
A-CodeQuality (#6)

Please note that all refactoring had already been done prior to this commit, incrementally.

PhireHandy committed:
Implemented an UndoCommand that deletes the last created Task.

PhireHandy committed:
Ready for release!

PhireHandy committed:
Final release for real...

PhireHandy committed:
Renamed the Screenshot

PhireHandy committed:
Tested and implemented a .jar file that works.

PhireHandy committed:
Added the correct Ui.png

PhireHandy committed:
Renamed Ui.png again

PhireHandy committed:
Set theme jekyll-theme-minimal

PhireHandy committed:
Renamed Ui.png again
